### PR TITLE
Ruff E through P

### PR DESCRIPTION
**Pull request by bandophahita**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -115,21 +123,21 @@ select = [
...
-#    "EM",  # flake8-errmsg
-#    "ERA",  # eradicate
-#    "F",  # Pyflakes
-#    "FA",  # flake8-future-annotations
+    "E",  # pycodestyle error
+    "EM",  # flake8-errmsg
+    "ERA",  # eradicate
+    "F",  # Pyflakes
+    "FA",  # flake8-future-annotations
 #    "FBT",  # flake8-boolean-trap
```
Comment (bandophahita) [inline pyproject.toml:131]: purposefully skipping FBT for this PR.  Will be addressed in it's own at the end.

File: pyproject.toml [hunk 2]
```diff
@@ -73,7 +73,7 @@ isort = {version = "*", optional = true}
 mypy = {version = "*", optional = true}
 pre-commit = {version = "*", optional = true}
 pytest = {version = "*", optional = true}
-ruff = {version = "*", optional = true}
+ruff = {version = ">=0.2.0", optional = true}
```
Comment (perrygoy) [inline pyproject.toml:76]: What necessitated this change?

File: tests/test_questions.py [hunk 3]
```diff
@@ -168,7 +170,7 @@ def test_caught_exception_annotation(self) -> None:
         e = Element(TARGET)
 
         annotation = e.__annotations__["caught_exception"]
-        assert annotation == Optional[TargetingError]
+        assert annotation == "Optional[TargetingError]"
```
Comment (perrygoy) [inline tests/test_questions.py:173]: Interesting, i'm assuming this is from the `from __future__ import annotations` change?

File: tests/test_resolutions.py [hunk 4]
```diff
@@ -80,7 +91,10 @@ def test_descriptions(self) -> None:
         _assert_descriptions(IsClickable(), element, expected)
 
     def test_type_hint(self) -> None:
-        assert_matcher_annotation(IsClickable())
+        ic = IsClickable()
+        annotation = ic.__annotations__["matcher"]
+        assert annotation == "IsClickableElement"
+        assert type(ic.matcher) == IsClickableElement
```
Comment (perrygoy) [inline tests/test_resolutions.py:97]: Nice updates to these tests 👍

File: tests/test_target.py [hunk 5]
```diff
@@ -86,17 +89,14 @@ def test_can_be_indexed() -> None:
 
 
 def test_locator_tuple_size() -> None:
-    with pytest.raises(ValueError) as excinfo:
-        Target("test").located((By.ID, "foo", "baz"))  # type: ignore
-    assert "locator tuple length should be 2" in f"{excinfo.value}"
+    with pytest.raises(ValueError, match="locator tuple length should be 2"):
+        Target("test").located((By.ID, "foo", "baz"))  # type: ignore[arg-type]
```
Comment (perrygoy) [inline tests/test_target.py:93]: I like these changes, too!

**Review by perrygoy — APPROVED** [review]
Some small questions for clarity and/or discussion, no notes! Thanks for the work, Marcel!

File: pyproject.toml (hunk 2 above)
Comment (bandophahita) [reply pyproject.toml:76]: The new rule changes in the pyproject.toml occurred at that version.
`[tool.ruff.lint.***]`

File: tests/test_questions.py (hunk 3 above)
Comment (bandophahita) [reply tests/test_questions.py:173]: Yep.  All annotations are strings when using future.

**Pull request merged**